persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -33,8 +34,24 @@ class MessageDAOImpl(private val queries: MessagesQueries) : MessageDAO {
 
     override suspend fun deleteAllMessages() = queries.deleteAllMessages()
 
-    override suspend fun insertMessage(message: MessageEntity) = queries.transaction { insertInDB(message) }
+    override suspend fun insertMessage(
+        message: MessageEntity,
+        updateConversationReadDate: Boolean,
+        updateConversationModifiedDate: Boolean,
+        updateConversationNotificationsDate: Boolean
+    ) {
+        queries.transaction {
+            if (updateConversationReadDate)
+                conversationsQueries.updateConversationReadDate(message.date, message.conversationId)
+            insertInDB(message)
+            if (updateConversationModifiedDate)
+                conversationsQueries.updateConversationModifiedDate(message.date, message.conversationId)
+            if (updateConversationNotificationsDate)
+                conversationsQueries.updateConversationNotificationsDate(message.date, message.conversationId)
```

Comment:
Could we use some `{ }` here ? I think the `insertInDB(message)` may get lost while reading at first sight.